addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf

```diff
@@ -5,188 +5,172 @@
+        TRACE_2("No valid wounds",_damage,_typeOfDamage);
+        continue
+    };
 
-            private _bodyPartNToAdd = if (_isSelectionSpecific) then {_bodyPartN} else {selectRandomWeighted _damageSelectionArray};
+    for "_i" from 1 to _nWounds do {
 
-            _bodyPartDamage set [_bodyPartNToAdd, (_bodyPartDamage select _bodyPartNToAdd) + _woundDamage];
-            _bodyPartVisParams set [[1,2,3,3,4,4] select _bodyPartNToAdd, true]; // Mark the body part index needs updating
+        // Select the injury we are going to add
+        selectRandomWeighted _weightedWoundTypes params ["_woundTypeToAdd", "", "_dmgMultiplier", "_bleedMultiplier", "_sizeMultiplier", "_painMultiplier", "_fractureMultiplier"];
+        GVAR(woundDetails) get _woundTypeToAdd params ["","_injuryBleedingRate","_injuryPain","_causeLimping","_causeFracture"];
+        private _woundClassIDToAdd = GVAR(woundClassNames) find _woundTypeToAdd;
+    
+        // Add a bit of random variance to wounds
+        private _woundDamage = _dmgPerWound * _dmgMultiplier * random [0.9, 1, 1.1];
 
+        _bodyPartDamage set [_bodyPartNToAdd, (_bodyPartDamage select _bodyPartNToAdd) + _woundDamage];
+        _bodyPartVisParams set [[1,2,3,3,4,4] select _bodyPartNToAdd, true]; // Mark the body part index needs updating
 
-            // Damage to limbs/head is scaled higher than torso by engine
-            // Anything above this value is guaranteed worst wound possible
-            private _worstDamage = [2, 1, 4, 4, 4, 4] select _bodyPartNToAdd;
+        // Anything above this value is guaranteed worst wound possible
+        private _worstDamage = 2;
 
-            // More wounds means more likely to get nasty wound
-            private _countModifier = 1 + random(_i - 1);
+        // Config specifies bleeding and pain for worst possible wound
+        // Worse wound correlates to higher damage, damage is not capped at 1
+        private _woundSize = linearConversion [0.1, _worstDamage, _woundDamage * _sizeMultiplier, 0.25, 1, true];
+    
+        private _pain = _woundSize * _painMultiplier * _injuryPain;
+        _painLevel = _painLevel + _pain;
 
-            // Config specifies bleeding and pain for worst possible wound
-            // Worse wound correlates to higher damage, damage is not capped at 1
-            private _bleedModifier = linearConversion [0.1, _worstDamage, _woundDamage * _countModifier, 0.25, 1, true];
-            private _painModifier = (_bleedModifier * random [0.7, 1, 1.3]) min 1; // Pain isn't directly scaled to bleeding
+        private _bleeding = _woundSize * _bleedMultiplier * _injuryBleedingRate;
 
-            private _bleeding = _injuryBleedingRate * _bleedModifier;
-            private _pain = _injuryPain * _painModifier;
-            _painLevel = _painLevel + _pain;
+        // wound category (minor [0.25-0.5], medium [0.5-0.75], large [0.75+])
+        private _category = floor linearConversion [0.25, 0.75, _woundSize, 0, 2, true];
```

Comment:
On the same topic I think these thresholds could be improved. As it is, the smallest "large" wound `0.75` is only 50% bigger than the largest "small" wound `0.499` which makes these terms less useful for gameplay.
I'd like to change it to an exponential scale like:
- minor: 0.125 - 0.25
- medium: 0.25 - 0.5
- large: 0.5-1

Which results in medium wounds being about twice as bad as minors, and large being about twice as bad as medium. The exact scale factor depends on what the minimum value is, something like `0.216-0.36-0.6-1` would work as well. This would obviously result in more wounds being categorised as large, but I think this would be a more honest representation of how dangerous those wounds are.